src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMultiFileCompilationModuleGroup.cs

```diff
@@ -0,0 +1,132 @@
+
+        public ReadyToRunMultiFileCompilationModuleGroup(TypeSystemContext context, IEnumerable<ModuleDesc> compilationModuleSet)
+        {
+            _compilationModuleSet = new HashSet<ModuleDesc>(compilationModuleSet);
+
+            // The fake assembly that holds compiler generated types is part of the compilation.
+            _compilationModuleSet.Add(context.GeneratedAssembly);
+        }
+
+        public sealed override bool ContainsType(TypeDesc type)
+        {
+            if (type is EcmaType ecmaType)
+            {
+                return IsModuleInCompilationGroup(ecmaType.EcmaModule);
+            }
+            if (type is InstantiatedType instantiatedType)
+            {
+                return ContainsType(instantiatedType.GetTypeDefinition());
+            }
+            return true;
+        }
+
+        public sealed override bool ContainsTypeDictionary(TypeDesc type)
+        {
+            return ContainsType(type);
+        }
+
+        public sealed override bool ContainsMethodBody(MethodDesc method, bool unboxingStub)
+        {
+            if (method.HasInstantiation)
+                return true;
+
+            return ContainsType(method.OwningType);
+        }
+
+        public sealed override bool ContainsMethodDictionary(MethodDesc method)
+        {
+            Debug.Assert(method.GetCanonMethodTarget(CanonicalFormKind.Specific) != method);
+            return ContainsMethodBody(method, false);
+        }
+
+        public sealed override ExportForm GetExportTypeForm(TypeDesc type)
+        {
+            return ExportForm.None;
+        }
+
+        public sealed override ExportForm GetExportTypeFormDictionary(TypeDesc type)
+        {
+            return ExportForm.None;
+        }
+
+        public sealed override ExportForm GetExportMethodForm(MethodDesc method, bool unboxingStub)
+        {
+            return ExportForm.None;
+        }
+
+        public override ExportForm GetExportMethodDictionaryForm(MethodDesc method)
+        {
+            return ExportForm.None;
+        }
+
+        private bool IsModuleInCompilationGroup(EcmaModule module)
+        {
+            return _compilationModuleSet.Contains(module);
+        }
+
+        public sealed override bool IsSingleFileCompilation
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public sealed override bool ShouldReferenceThroughImportTable(TypeDesc type)
+        {
+            return false;
+        }
+
+        public override bool CanHaveReferenceThroughImportTable
+        {
+            get
+            {
+                return false;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Represents a non-leaf multifile compilation group where types contained in the group are always fully expanded.
+    /// </summary>
+    public class ReadyToRunMultiFileSharedCompilationModuleGroup : ReadyToRunMultiFileCompilationModuleGroup
```

Comment:
So we'll want these two classes merged down into one, with the overrides from `ReadyToRunMultiFileSharedCompilationModuleGroup` replacing the base implementations in `ReadyToRunMultiFileCompilationModuleGroup`